packages/opentelemetry-plugin-xml-http-request/src/xhr.ts

```diff
@@ -0,0 +1,371 @@
+    if (!propagateTraceHeaderUrls) {
+      return;
+    }
+
+    if (
+      typeof propagateTraceHeaderUrls === 'string' ||
+      propagateTraceHeaderUrls instanceof RegExp
+    ) {
+      propagateTraceHeaderUrls = [propagateTraceHeaderUrls];
+    }
+
+    const url = (span as tracing.Span).name;
+    const spanUrl = parseUrl(url);
+    if (spanUrl.origin !== window.location.origin) {
+      this._logger.warn('Cannot set headers on different origin');
+      return;
+    }
+
+    for (const propagateTraceHeaderUrl of propagateTraceHeaderUrls) {
+      if (sameOriginOrUrlMatches(url, propagateTraceHeaderUrl)) {
+        const b3Format = new B3Format();
+        const carrier: { [key: string]: unknown } = {};
+        b3Format.inject(span.context(), 'B3Format', carrier);
+
+        Object.keys(carrier).forEach(key => {
+          xhr.setRequestHeader(key, String(carrier[key]));
+        });
+      }
+    }
+  }
+
+  /**
+   * Adds span network events
+   * @param span
+   */
+  private _addSpanNetworkEvents(span: tracing.Span) {
+    const resources: PerformanceResourceTiming[] = otperformance.getEntriesByType(
+      'resource'
+    ) as PerformanceResourceTiming[];
+    const resource = getResource(
+      span,
+      resources,
+      this._usedResources[span.name]
+    );
+
+    if (resource) {
+      this._markResourceAsUsed(resource);
+      addSpanNetworkEvent(span, PTN.FETCH_START, resource);
+      addSpanNetworkEvent(span, PTN.DOMAIN_LOOKUP_START, resource);
+      addSpanNetworkEvent(span, PTN.DOMAIN_LOOKUP_END, resource);
+      addSpanNetworkEvent(span, PTN.CONNECT_START, resource);
+      addSpanNetworkEvent(span, PTN.SECURE_CONNECTION_START, resource);
+      addSpanNetworkEvent(span, PTN.CONNECT_END, resource);
+      addSpanNetworkEvent(span, PTN.REQUEST_START, resource);
+      addSpanNetworkEvent(span, PTN.RESPONSE_START, resource);
+      addSpanNetworkEvent(span, PTN.RESPONSE_END, resource);
+    }
+  }
+
+  /**
+   * Removes the previous information about span.
+   * This might happened when the same xhr is used again.
+   * @param xhr
+   * @private
+   */
+  private _cleanPreviousSpanInformation(xhr: XMLHttpRequestWrapped) {
+    const existingSpanId = xhr.__OT_SPAN_ID;
+    if (existingSpanId) {
+      const callbackToRemoveEvents = this._callbackToRemoveEvents[
+        existingSpanId
+      ];
+      if (callbackToRemoveEvents) {
+        callbackToRemoveEvents();
+      }
+    }
+  }
+
+  /**
+   * Creates a new span when method "open" is called
+   * @param xhr
+   * @param url
+   * @param method
+   * @private
+   */
+  private _createSpan(
+    xhr: XMLHttpRequestWrapped,
+    url: string,
+    method: string
+  ): tracing.Span | undefined {
+    if (isUrlIgnored(url, this._config.ignoreUrls)) {
+      this._logger.debug('ignoring span as url matches ignored url');
+      return;
+    }
+
+    const currentSpan = this._tracer.startSpan(url, {
+      parent: this._tracer.getCurrentSpan() || undefined,
```

Comment:
This should no longer be required
```suggestion
      parent: this._tracer.getCurrentSpan(),
```